connectors/sources/google_cloud_storage.py

```diff
@@ -242,12 +288,12 @@ async def fetch_blobs(self, buckets):
             Dictionary: Contains the list of fetched blobs from Google Cloud Storage.
         """
         for bucket in buckets.get("items", []):
-            async for blob in self._api_call(
+            async for blob in self.google_storage_client._api_call(
                 resource="objects",
                 method="list",
                 full_response=True,
                 bucket=bucket["id"],
-                userProject=self.user_project_id,
+                userProject=self.google_storage_client.user_project_id,
```

Comment:
Same as above, we probably don't need that and use the field inside the client class?